src/dlsproto/client/request/internal/GetRange.d

```diff
@@ -398,11 +398,49 @@ private scope class GetRangeHandler
 
     private /* scope */ class RecordStream
     {
-        /// The acquired buffer to store a batch of records.
-        private void[]* batch_buffer;
+        /// Double buffer structure
+        private struct DoubleBuffer
+        {
+            /// Acquired buffers to store batches of records. Reader always
+            /// appends to the back one and the RecordStream always uses the
+            /// front one.
+            private void[]*[2] batch_buffers;
+
+            /// Index of the batch we're currently iterating over
+            private int current_buffer;
+
+            /// Initializes the double buffer
+            void init (void[]* delegate() getVoidBuffer)
```

Comment:
`lazy void[]*` if you want to be cool ;)

Reply:
Hmm, I think the way it is consistent with rest of the swarm - we're passing acquires as delegates.